app/plugin.go

```diff
@@ -76,56 +77,69 @@ func (a *App) SyncPluginsActiveState() {
 			return
 		}
 
+		var wg sync.WaitGroup
 		// Deactivate any plugins that have been disabled.
 		for _, plugin := range availablePlugins {
-			// Determine if plugin is enabled
-			pluginId := plugin.Manifest.Id
-			pluginEnabled := false
-			if state, ok := config.PluginStates[pluginId]; ok {
-				pluginEnabled = state.Enable
-			}
+			wg.Add(1)
+			go func(plugin *model.BundleInfo) {
+				defer wg.Done()
+				// Determine if plugin is enabled
+				pluginId := plugin.Manifest.Id
+				pluginEnabled := false
+				if state, ok := config.PluginStates[pluginId]; ok {
+					pluginEnabled = state.Enable
+				}
 
-			// If it's not enabled we need to deactivate it
-			if !pluginEnabled {
-				deactivated := pluginsEnvironment.Deactivate(pluginId)
-				if deactivated && plugin.Manifest.HasClient() {
-					message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_PLUGIN_DISABLED, "", "", "", nil)
-					message.Add("manifest", plugin.Manifest.ClientManifest())
-					a.Publish(message)
+				// If it's not enabled we need to deactivate it
+				if !pluginEnabled {
+					deactivated := pluginsEnvironment.Deactivate(pluginId)
+					if deactivated && plugin.Manifest.HasClient() {
+						message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_PLUGIN_DISABLED, "", "", "", nil)
+						message.Add("manifest", plugin.Manifest.ClientManifest())
+						a.Publish(message)
+					}
 				}
-			}
+			}(plugin)
+
 		}
+		wg.Wait()
```

Comment:
Do we need to wait here, or can we immediately proceed to activate plugins as well?

Reply:
Yes we can proceed. Made the change.